roiextractors/imagingextractor.py

```diff
@@ -58,6 +58,38 @@ def get_dtype(self) -> DtypeType:
     def get_video(self, start_frame: int = None, end_frame: int = None, channel: int = 0) -> NumpyArray:
         return self.get_frames(range(start_frame, end_frame), channel)
 
+    def frame_to_time(self, frame: int):
+        '''This function converts a user-inputted frame index to a time with units of seconds.
+
+        Parameters
+        ----------
+        frame: float
+            The frame to be converted to a time
+
+        Returns
+        -------
+        time: float
+            The corresponding time in seconds
+        '''
+        # Default implementation
+        return frame / self.get_sampling_frequency()
+
+    def time_to_frame(self, time):
+        '''This function converts a user-inputted time (in seconds) to a frame index.
```

Comment:
handle irregularly sampled data. Does this exist for ophys? I'm not sure

Reply:
I would keep regularly sampled data for now. Let's have another PR to handle that